src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,24 @@
+module Algebra.Graph.Bipartite.AdjacencyMap (
+    AdjacencyMap, leftToRight, rightToLeft,
+    consistent,
+    ) where
+
+import qualified Data.Map.Strict as Map
+import qualified Data.Set        as Set
+
+data AdjacencyMap a b = BAM {
+    leftToRight :: Map.Map a (Set.Set b),
+    rightToLeft :: Map.Map b (Set.Set a)
+} deriving Eq
+
+consistent :: (Ord a, Ord b) => AdjacencyMap a b -> Bool
+consistent (BAM lr rl) = oneWayConstistent lr rl && oneWayConstistent rl lr
+
+internalEdgeList :: Map.Map a (Set.Set b) -> [(a, b)]
+internalEdgeList lr = [ (u, v) | (u, vs) <- Map.toAscList lr, v <- Set.toAscList vs ]
+
+hasEdge :: (Ord a, Ord b) => Map.Map a (Set.Set b) -> a -> b -> Bool
```

Comment:
It would be best to make this function similar to the API of other modules:

```haskell
hasEdge :: (Ord a, Ord b) => a -> b -> AdjacencyMap a b -> Bool
```

Reply:
@snowleopard It was first designed to be used in `consistent`. Now that we don't need it to check consistency, I will change the signature.